src/grammar_optimiser/analyze_compressions.jl

```diff
@@ -0,0 +1,153 @@
+"""
+    generate_stats(d, compressed_AST)
+
+Compression Analysis. Analyzes 1 AST to see how many times each compression was used.
+# Arguments
+- `d::Dict`: the global dictionary (key: node_id, value: namedTuple(compressiond_id, parent_id, child_nr, type, [children]))
+- `compressed_AST::Vector{String}`: a list of assign-statements ["assign(A, X)", assign(B, Y), ...]
+# Result 
+- `c_info::Dict{Int64, NamedTuple{(:size, :occurences), <:Tuple{Int64,Int64}}}`: an dict(key: compression_id, value: Tuple(size, # occurences))) 
+"""
+function generate_stats(d, compressed_AST)
```

Comment:
- Please add types to your header. You have it in the docstring already.
- please rename to sth like `generate_compression_stats` or so.
- could you also clean up the code a bit? i.e. give variables proper names so we still understand what they mean in 6 months from now. Also remove all code that is commented out.